qiskit/providers/aer/jobs/utils.py

```diff
@@ -126,12 +126,13 @@ def _check_custom_instruction(experiments, optypes=None):
     if optypes is not None:
         # Optypes store class names as strings
         return any(
-            "Save" in name or "Snapshot" in name
-            for optype in optypes for name in optype
+            {"SaveData", "Snapshot"}.intersection(optype)
+            for optype in optypes
         )
```

Comment:
Do these come from `Qobj` in this case?  If so, then we really _need_ to make sure we're storing `optype` as a `set`, not a `list` to avoid the nasty iteration penalties here.